modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/Violations.tsx

```diff
@@ -0,0 +1,255 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import './Violations.scss';
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import {ClayCheckbox} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayList from '@clayui/list';
+import React, {useState} from 'react';
+
+import Rule from './Rule';
+import {TYPES, useFilteredViolationsDispatch} from './useFilteredViolations';
+
+import type {ImpactValue, Result} from 'axe-core';
+
+declare var Liferay: {
+	Language: {
+		get(value: string): string;
+	};
+	Util: {
+		sub(...value: string[]): string;
+	};
+};
+
+const IMPACT_FILTER_OPTIONS = [
+	{label: Liferay.Language.get('critical'), value: 'critical'},
+	{label: Liferay.Language.get('serious'), value: 'serious'},
+	{label: Liferay.Language.get('moderate'), value: 'moderate'},
+	{label: Liferay.Language.get('minor'), value: 'minor'},
+];
+
+const CATEGORY_FILTER_OPTIONS = [
+	{label: 'WCAG 2.0A', value: 'wcag2a'},
+	{label: 'WCAG 2.0 Level AA', value: 'wcag2aa'},
+	{label: 'WCAG 2.1 Level A', value: 'wcag21a'},
+	{label: 'WCAG 2.1 Level AA', value: 'wcag21aa'},
+	{label: 'Best Practices', value: 'best-practice'},
+];
+
+type ViolationsFilterProps = {
+	selectedCategories: Array<String>;
+	selectedImpact: Array<ImpactValue>;
+};
+
+function ViolationsFilter({
+	selectedCategories,
+	selectedImpact,
+}: ViolationsFilterProps) {
+	const [dropdownExpanded, setDropdownExpanded] = useState(false);
+	const dispatch = useFilteredViolationsDispatch();
+
+	return (
+		<>
+			<ClayDropDown
+				active={dropdownExpanded}
+				closeOnClickOutside
+				onActiveChange={setDropdownExpanded}
+				trigger={
+					<ClayButtonWithIcon
+						aria-label={Liferay.Language.get(
+							'open-violations-filter'
+						)}
+						displayType="unstyled"
+						small
+						symbol="filter"
+					/>
+				}
+			>
+				<ClayDropDown.ItemList>
+					<ClayDropDown.Group
+						header={Liferay.Language.get('filter-by-impact')}
+					>
+						{IMPACT_FILTER_OPTIONS.map((item) => {
+							const existsImpact =
+								selectedImpact.findIndex(
+									(element) => element === item.value
+								) !== -1;
+
+							return (
+								<ClayDropDown.Section
+									active={existsImpact}
+									key={item.value}
+								>
+									<ClayCheckbox
+										checked={existsImpact}
+										data-testid={item.value}
+										label={item.label}
+										onChange={() => {
+											if (existsImpact) {
+												dispatch({
+													payload: {
+														value: item.value,
+													},
+													type: TYPES.IMPACT_REMOVE,
+												});
+
+												return;
+											}
+
+											dispatch({
+												payload: {
+													value: item.value,
+												},
+												type: TYPES.IMPACT_ADD,
+											});
+										}}
+									></ClayCheckbox>
```

Comment:
Just to remind you https://github.com/liferay-frontend/liferay-portal/pull/1076#discussion_r638032168